feathr_project/feathr/secrets/akv_client.py

```diff
@@ -15,8 +15,10 @@ def get_akv_secret(self, secret_name: str):
                 credential=DefaultAzureCredential()
             )
         try:
+            # replace '_' with '-' since Azure Key Vault doesn't support it
```

Comment:
Shall we split it into a separate function, say get_akv_name_from_config_name(): for better maintenance?
Also, we could add a note in docs to make sure that key vault name and config name always follow the same naming rules, e.g., in https://github.com/linkedin/feathr/blob/main/docs/how-to-guides/azure-deployment.md